server/compose/service/record.go

```diff
@@ -1265,6 +1267,32 @@ func (svc record) delete(ctx context.Context, namespaceID, moduleID, recordID ui
 	return del, nil
 }
 
+func (svc record) undelete(ctx context.Context, namespaceID, moduleID, recordID uint64) (undel *types.Record, err error) {
+    if namespaceID == 0 {
+        return nil, RecordErrInvalidNamespaceID()
+    }
+    if moduleID == 0 {
+        return nil, RecordErrInvalidModuleID()
+    }
+    if recordID == 0 {
+        return nil, RecordErrInvalidID()
+    }
```

Comment:
Shouldn't this validation be done in the exported undelete function? Perhaps worth considering having the whole undelete logic in the same function